src/App/Validator/User/Setting/Update.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * Ushahidi User Setting Validator
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2018 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Validator\User\Setting;
+
+use Ushahidi\Core\Entity;
+use Ushahidi\Core\Entity\UserRepository;
+use Ushahidi\Core\Tool\Validator;
+
+class Update extends Validator
+{
+	protected $user_repo;
+	protected $default_error_source = 'user_setting';
+
+
+	public function __construct(UserRepository $user_repo)
+	{
+		$this->user_repo = $user_repo;
+	}
+
+	protected function getRules()
+	{
+		return [
+			'user_id' => [
+				['digit'],
+				[[$this->user_repo, 'exists'], [':value']],
+			],
+			'config_key' => [
```

Comment:
Add a list of allowed keys? or maybe we don't even care

Reply:
Right now we don't have a list but I'll add it as a comment.